From fe70bb849e6c3d6f5882b7c64ed14757be53f81f Mon Sep 17 00:00:00 2001
From: Johan Hovold <jhovold@gmail.com>
Date: Mon, 29 Oct 2012 10:56:26 +0100
Subject: [PATCH 09/12] USB: ftdi_sio: use generic chars_in_buffer
References: bnc#956375
Git-Commit: 755b6040fa62eab9d9105359cd5884910eef2df4
Patch-Mainline: v3.8

Use generic chars_in_buffer rather than copying it's implementation.

Signed-off-by: Johan Hovold <jhovold@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/usb/serial/ftdi_sio.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/drivers/usb/serial/ftdi_sio.c b/drivers/usb/serial/ftdi_sio.c
index 5153545..d179bc0 100644
--- a/drivers/usb/serial/ftdi_sio.c
+++ b/drivers/usb/serial/ftdi_sio.c
@@ -2166,16 +2166,11 @@ static int ftdi_chars_in_buffer(struct tty_struct *tty)
 {
 	struct usb_serial_port *port = tty->driver_data;
 	struct ftdi_private *priv = usb_get_serial_port_data(port);
-	unsigned long flags;
 	int chars;
 	unsigned char *buf;
 	int ret;
 
-	/* Check software buffer (code from
-	 * usb_serial_generic_chars_in_buffer()) */
-	spin_lock_irqsave(&port->lock, flags);
-	chars = kfifo_len(&port->write_fifo) + port->tx_bytes;
-	spin_unlock_irqrestore(&port->lock, flags);
+	chars = usb_serial_generic_chars_in_buffer(tty);
 
 	/* Check hardware buffer */
 	switch (priv->chip_type) {
-- 
2.1.4

